### ISSUE TITLE
_draw_polygon does not work with colors array (only single color works)

### ISSUE THREAD

**Issue by toger5** [godotengine/godot#9147]
This is how the stylboxes are drawn now. 
https://youtu.be/1UTMRCvx6Mg
They are effected by the color I choose in the inspector.

My draw poly command is structured like this:
arrayOfVerts ArrayOfColors ArrayOfIndices
the arrayOfColors and the ArrayOfVerts are the same length because that's how the _draw_poly function I used was implemented, one color for one vert (not per index).
Is that the issue.
Why it the color chooses in the Insoctor affecting it?

And a screenshot for the lazy ones who won't checkout the youtube video:
<img width="1680" alt="screen shot 2017-06-13 at 01 22 35" src="https://user-images.githubusercontent.com/16718859/27072912-d04b016a-4fd6-11e7-99fb-1d8fac15174e.png">

**Comment by toger5** [comment]
I can't even get a really simple triangle array working with multiple colors
This does not get drawn...
```cpp
	Vector<Point2> vertsT;
	Vector<int> indicesT;
	Vector<Color> colorsT;
	vertsT.push_back(p_rect.position);
	vertsT.push_back(p_rect.position + Vector2(p_rect.size.x,0));
	vertsT.push_back(p_rect.position + p_rect.size);
	indicesT.push_back(0);
	indicesT.push_back(2);
	indicesT.push_back(1);
	Color cT = Color(1, 0, 0);
	colorsT.push_back(cT);
	colorsT.push_back(cT);
	colorsT.push_back(cT);
	vs->canvas_item_add_triangle_array(p_canvas_item, indicesT, vertsT, colorsT);
```

this works though...
```
	Vector<Point2> vertsT;
	Vector<int> indicesT;
	Vector<Color> colorsT;
	vertsT.push_back(p_rect.position);
	vertsT.push_back(p_rect.position + Vector2(p_rect.size.x,0));
	vertsT.push_back(p_rect.position + p_rect.size);
	indicesT.push_back(0);
	indicesT.push_back(2);
	indicesT.push_back(1);
	Color cT = Color(1, 0, 0);
	colorsT.push_back(cT);
//	colorsT.push_back(cT); comment out the additional colors for the color array
//	colorsT.push_back(cT);
	vs->canvas_item_add_triangle_array(p_canvas_item, indicesT, vertsT, colorsT);
```

**Comment by groud** [comment]
If I look at the code in (https://github.com/godotengine/godot/blob/master/servers/visual/visual_server_canvas.cpp) I find that :
```
ERR_FAIL_COND(!p_colors.empty() && p_colors.size() != ps && p_colors.size() != 1);
ERR_FAIL_COND(!p_uvs.empty() && p_uvs.size() != ps);
```
I think there is an error: the `&&` should be replaced by a `||`, am I right ?

**Comment by toger5** [comment]
@reduz Do you have any idea what the cause could be?
this lines of code make it not work:
```cpp
	} else {

		glBufferSubData(GL_ARRAY_BUFFER, buffer_ofs, sizeof(Color) * p_vertex_count, p_colors);
		glEnableVertexAttribArray(VS::ARRAY_COLOR);
		glVertexAttribPointer(VS::ARRAY_COLOR, 4, GL_FLOAT, false, sizeof(Color), ((float *)0) + buffer_ofs);
		buffer_ofs += sizeof(Color) * p_vertex_count;
	}
```

**Comment by toger5** [comment]
if I replace this: 		
```cpp
glVertexAttribPointer(VS::ARRAY_COLOR, 4, GL_FLOAT, false, sizeof(Color), ((float *)0) + buffer_ofs);
```
with 
```cpp
glVertexAttribPointer(VS::ARRAY_COLOR, 4, GL_FLOAT, false, sizeof(Color), (float *)buffer_ofs);
```
it works

**Comment by toger5** [comment]
fixed with recent commit by reduz

**Comment by bojidar-bg** [comment]
For the record, fixed by 2b62872547f11debfe45a4eb697b36ef05fbf2eb.
